Packs/TOPdesk/Integrations/TOPdesk/TOPdesk.py

```diff
@@ -0,0 +1,862 @@
+    )
+
+
+''' FETCH & MIRRORING COMMANDS'''
+
+
+def fetch_incidents(client: Client,
+                    last_run: Dict[str, Any],
+                    demisto_params: Dict[str, Any]) -> Tuple[Dict[str, Any], List[Dict[str, Any]]]:
+    """Fetches incidents from TOPdesk."""
+
+    first_fetch_datetime = dateparser.parse(demisto_params.get('first_fetch', '3 days'))
+    last_fetch = last_run.get('last_fetch', None)
+
+    if not last_fetch:
+        if first_fetch_datetime:
+            last_fetch_datetime = first_fetch_datetime
+        else:
+            raise Exception("Could not find last fetch time.")
+    else:
+        last_fetch_datetime = dateparser.parse(last_fetch)
+
+    latest_created_time = last_fetch_datetime
+    incidents: List[Dict[str, Any]] = []
+
+    topdesk_incidents = get_incidents_with_pagination(client=client,
+                                                      max_fetch=int(demisto_params.get('max_fetch', 10)),
+                                                      query=demisto_params.get('query', None),
+                                                      modification_date_start=demisto_params.get(
+                                                          'modification_date_start', None),
+                                                      modification_date_end=demisto_params.get(
+                                                          'modification_date_end', None))
+
+    for topdesk_incident in topdesk_incidents:
+        if topdesk_incident.get('creationDate', None):
+            creation_datetime = dateparser.parse(topdesk_incident['creationDate'])
+            incident_created_time = creation_datetime
+        else:
+            incident_created_time = last_fetch_datetime
+        if int(last_fetch_datetime.timestamp()) < int(incident_created_time.timestamp()):
+            labels: List[Dict[str, Any]] = []
+            for topdesk_incident_field, topdesk_incident_value in topdesk_incident.items():
+                if isinstance(topdesk_incident_value, str):
+                    labels.append({
+                        'type': topdesk_incident_field,
+                        'value': topdesk_incident_value
+                    })
+                else:
+                    labels.append({
+                        'type': topdesk_incident_field,
+                        'value': json.dumps(topdesk_incident_value)
+                    })
+
+            incidents.append({
+                'name': f"TOPdesk incident {topdesk_incident['number']}",
+                'details': json.dumps(topdesk_incident),
+                'occurred': incident_created_time.strftime(DATE_FORMAT),
+                'rawJSON': json.dumps(topdesk_incident),
+                'labels': labels
+            })
+        if latest_created_time.timestamp() < incident_created_time.timestamp():
+            latest_created_time = incident_created_time
+
+    return {'last_fetch': latest_created_time.strftime(DATE_FORMAT)}, incidents
```

Comment:
by moving to str here you are losing millisecond info and this can result in duplicated incidents - do we know that topdesk does not use millisec? if not then you should keep the data in a different format like the datetime object or numeric timestamp.

Reply:
I redid this so the millisecond and timezone will be taken into account